### ISSUE TITLE
istio/proxy postsubmit failed

### ISSUE THREAD

**Issue by bianpengyuan** [istio/istio#18977]
https://prow.istio.io/view/gcs/istio-prow/logs/release_proxy_postsubmit/1

```
++ lsb_release -c -s
+ UBUNTU_RELEASE=bionic
+ [[ bionic == \x\e\n\i\a\l ]]
+ echo 'Must run on Ubuntu 16.04 (Xenial).'
Must run on Ubuntu 16.04 (Xenial).
+ exit 1
Makefile.core.mk:104: recipe for target 'push_release' failed
make: *** [push_release] Error 1
```

The new builder image we use is bionic. @PiotrSikora Is there any reason must be on xenial?

**Comment by PiotrSikora** [comment]
That's what we were using historically. I don't think there is any reason to do that, but the check was in place (and should stay there for a new base distro) in order to catch unintentional changes.

**Comment by lambdai** [comment]
https://unix.stackexchange.com/questions/9705/can-new-glibc-versions-be-used-with-an-old-kernel

Docker doesn't help much in terms of GLIBC incompatible

bionic = 18.04 = glibc 2.27
xenial = 16.04 = glibc 2.23

**Comment by bianpengyuan** [comment]
I think Istio docker images are built from bionic, so glibc compatibility should not be a problem.

**Comment by lizan** [comment]
Note binary built on bionic doesn't run on xenial due to glibc version, but vice versa.

**Comment by lizan** [comment]
But are we on RBE for post submit as well? If so it doesn't matter.

**Comment by bianpengyuan** [comment]
I think @PiotrSikora temporarily move postsubmit off rbe, due to some permission issue. 

The binary built from postsubmit is used for istio-proxy docker image, which is also based on bionic, so it should be fine, no?

**Comment by lizan** [comment]
@bianpengyuan it will likely break most of VM deployments

**Comment by lizan** [comment]
The compatibility bit is: binary built on xenial runs on glibc >= 2.18, which is lesser than most of recent Linux distro, but binary built on bionic requires glibc >= 2.26 IIRC.

**Comment by bianpengyuan** [comment]
Does user use proxy binary tarball pushed from istio/proxy repo to deploy on VM? First time know about this, I thought it is just used for proxy docker image..

**Comment by lambdai** [comment]
The release in istio/proxy repo doesn't contain compiled binary. Are you actually referring to the tar in istio/istio release?

**Comment by lizan** [comment]
Do we still release istio-sidecar.deb? (at least docs says so) That's where the compatibility is broken.

**Comment by lambdai** [comment]
I am afraid so... https://gcsweb.istio.io/gcs/istio-release/releases/1.4.0/deb/

**Comment by bianpengyuan** [comment]
I see. Then I need to downgrade proxy builder image to 16.04 then to make post submit pass then.

**Comment by lizan** [comment]
And also binary built on Bionic doesn't run on distroless either, until distroless moved to debian 10 base (which is in progress)

**Comment by PiotrSikora** [comment]
Yes, please downgrade.

@lizan we're not on RBE for postsubmit, but that was only blocked by lack of `gcloud` in the RBE toolchain, which you fixed by now.

**Comment by lizan** [comment]
@PiotrSikora `container_push` is using `bazel run` so it is local so I doubt RBE with gcloud helps. You need [configure](https://github.com/bazelbuild/rules_docker#container_push-custom-client-configuration) docker toolchain.

**Comment by kyessenov** [comment]
container_push does not use docker. It pushes the image directly without the daemon.

**Comment by lizan** [comment]
@kyessenov no, oci_push pushes directly but container_push uses docker.

**Comment by kyessenov** [comment]
That's interesting. I was pretty sure that it didn't, since it passed on post-submit where there is no docker set-up. I would argue we don't want docker since it's insecure in prow.

**Comment by lizan** [comment]
@kyessenov there's obviously docker set-up in post-submit, otherwise `gcloud auth configure-docker` will put big WARNING about docker is not available.

**Comment by lizan** [comment]
@kyessenov https://github.com/istio/test-infra/blob/master/docker/prowbazel/entrypoint#L17

**Comment by kyessenov** [comment]
Alright. Can we use oci_push? It's pretty terrible that we have root in the container.

**Comment by lizan** [comment]
Not sure, it is pretty experimental and I have no idea whether it works well with gcr.

**Comment by bianpengyuan** [comment]
Postsubmit now passes by making rbe work https://prow.istio.io/view/gcs/istio-prow/logs/release_proxy_postsubmit/8. @kyessenov Please open another issue if you want to move envoy container push away from docker.
